src/openrct2/world/MapGen.cpp

```diff
@@ -134,46 +138,43 @@ void mapgen_generate_blank(mapgen_settings* settings)
 
 void mapgen_generate(mapgen_settings* settings)
 {
-    int32_t x, y, mapSize, floorTexture, wallTexture, waterLevel;
-
-    mapSize = settings->mapSize;
-    floorTexture = settings->floor;
-    wallTexture = settings->wall;
-    waterLevel = settings->water_level;
-
-    if (floorTexture == -1)
+    auto mapSize = settings->mapSize;
+    auto waterLevel = settings->water_level;
+    const auto selectedFloor = TerrainSurfaceObject::GetById(settings->floor);
+    std::string floorTexture = selectedFloor != nullptr ? std::string(selectedFloor->GetIdentifier()) : "";
```

Comment:
```suggestion
    std::string floorTexture = selectedFloor != nullptr ? std::string(selectedFloor->GetIdentifier()) : std::string();
```
I'd personally prefer doing this, but not everyone likes it